README.md

```diff
@@ -159,7 +159,103 @@ pub fn do_twist_command(&mut self) -> Result<(), TwistError> {
 }
 ```
 
-<!-- TODO: CLI, payloads, now, collector, jon's tracing example?, cli export -->
+### Generate Event Manifests and Definitions
+
+In the samples above, a macro is used to initialize a probe and to
+record events. Modality Probe's CLI has a subcommand that explores
+your code base for uses of these macros and turns those uses into what
+Modality calls a _Component_. A component has a name of its own, a
+list of probes (`probes.csv`), and a list of events
+(`events.csv`). Altogether, a component looks like this:
+
+```
+$ tree my-component
+├── Component.toml
+├── events.csv
+└── probes.csv
+```
+
+`manifest-gen` is the subcommand we use to do this. Use it like this:
+
+```shell
+$ modality-probe manifest-gen --component-name my-component -o my-component .
+```
+
+Now you'll have the `my-component` directory in your working directory
+and its contents should look like the example at the top of this
+section.
+
+Next, we'll want to generate the source code that gives those symbols
+we discussed in the code snippet examples in the previous section
+their definitions. To do that, we'll use `header-gen`:
+
+```shell
+$ modality-probe header-gen -o include/events.h my-component
+```
+
+This will deposit a file at `include/events.h` that contains
+definitions for each probe and event in the manifests in your
+component.
+
+If you're using Rust, you can tell the command to generate Rust
+instead using the `--lang` switch (C is the default):
+
+```shell
+$ modality-probe header-gen -o include/events.h -l rust my-component
```

Comment:
This line probably should use a Rust file instead of a C header file: `-o events.rs`